test/functional/buildAndPackage/src/net/adoptopenjdk/test/CudaEnabledTest.java

```diff
@@ -4,96 +4,125 @@
+                while ((oneLine = prtFileReader.readLine()) != null) {
+                    if (oneLine.contains("cudart")) {
+                        logger.info("Cuda-enabled indicator string \'cudart\' was found within this j9prt file: " + prtFile);
+                        foundCudart = true;
+                        break;
+                    }
+                }
+                prtFileReader.close();
+                if (foundCudart) {
+                    successes++;
+                    continue;
+                }
+            } catch (FileNotFoundException e) {
+                Assert.fail("A file that exists could not be found. This should never happen.");
+            } catch (Exception e) {
+                throw new Error(e);
             }
+
+            logger.info("Cuda-enabled indicator string \'cudart\' was not found within this j9prt file: " + prtFile);
         }
 
-        File prtDirObject = new File(prtLibDirectory);
-        Assert.assertTrue(prtDirObject.exists(), "Can't find the predicted location of the j9prt lib file. Expected location: " + prtLibDirectory);
-        
-        String[] prtLibDirectoryFiles = prtDirObject.list();
-        String prtFile = null;
-        for(int x = 0 ; x < prtLibDirectoryFiles.length ; x++) {
-        	if(prtLibDirectoryFiles[x].contains("j9prt")) {
-        	    prtFile = prtLibDirectory + "/" + prtLibDirectoryFiles[x];
-        	    break;
-        	}
+        logger.info("j9prt file scanning complete. Assessing results.");
+
+        if ((successes > 0) && (successes < j9prtFiles.size())) {
+            Assert.fail("Multiple j9prt files were found, some with cuda enabled and some without.");
         }
-        Assert.assertNotNull(prtFile,"Can't find the j9prt lib file in " + prtLibDirectory);
-        Assert.assertTrue((new File (prtFile)).exists(), "Found the prt file, but it doesn't exist. Tautology bug.");
-        Assert.assertTrue((new File (prtFile)).canRead(), "Found the prt file, but it can't be read. Likely a permissions bug.");
-        
-        //Stage 2: Iterate through the j9prt lib file to find "cudart".
-        //If we find it, then cuda functionality is enabled on this build.
-        try {
-            BufferedReader prtFileReader = new BufferedReader(new FileReader(prtFile));
-            String oneLine = "";
-            while ((oneLine = prtFileReader.readLine()) != null) {
-                if(oneLine.contains("cudart")) {
-                	logger.info("Test completed successfully.");
-                    return; //Success!
-                }
+
+        if (jdkPlatform.runsOn(OperatingSystem.WINDOWS)) {
+            if (jdkPlatform.runsOn(Architecture.X64)) {
+                Assert.assertTrue(successes == j9prtFiles.size(), "One or more of the j9prt files in this build is not cuda-enabled.");
+            } else {
+                Assert.assertTrue(successes == 0, "This build is cuda-enabled, but non-x64 Windows builds are not expected to be.");
```

Comment:
It's much better to use `assertEquals` than `assertTrue` where possible: the former yields messages that show both numbers making it clearer how the test failed.

Reply:
Fair enough. Will make that change now.